models/decam.py

```diff
@@ -0,0 +1,523 @@
+
+    def _make_new_section(self, section_id):
+        """
+        Make a single section for the DECam instrument.
+        The section_id must be a valid section identifier (Si or Ni, where i is an int in [1,31])
+
+        Returns
+        -------
+        section: SensorSection
+            A new section for this instrument.
+        """
+        (dx, dy) = self.get_section_offsets(section_id)
+        defective = section_id in { 'N30', 'S7' }
+        return SensorSection(section_id, self.name, size_x=2048, size_y=4096,
+                             offset_x=dx, offset_y=dy, defective=defective)
+
+    def get_ra_dec_for_section( self, exposure, section_id ):
+        if section_id not in self._chip_radec_off:
+            raise ValueError( f"Unknown DECam section_id {section_id}" )
+        return ( exposure.ra + self._chip_radec_off[section_id]['dra'] / math.cos( exposure.dec * math.pi / 180. ),
+                 exposure.dec + self._chip_radec_off[section_id]['ddec'] )
+
+    @classmethod
+    def _get_fits_hdu_index_from_section_id(cls, section_id):
+        """
+        Return the index of the HDU in the FITS file for the DECam files.
+        Since the HDUs have extension names, we can use the section_id directly
+        to index into the HDU list.
+        """
+        cls.check_section_id(section_id)
+        return section_id
+
+    @classmethod
+    def get_filename_regex(cls):
+        return [r'c4d.*\.fits']
+
+    @classmethod
+    def get_short_instrument_name(cls):
+        """
+        Get a short name used for e.g., making filenames.
+        """
+        return 'c4d'
+
+    @classmethod
+    def get_short_filter_name(cls, filter):
+        """
+        Return the short version of each filter used by DECam.
+        In this case we just return the first character of the filter name,
+        e.g., shortening "g DECam SDSS c0001 4720.0 1520.0" to "g".
+        """
+        return filter[0:1]
+
+    def find_origin_exposures( self, skip_exposures_in_database=True,
+                               minmjd=None, maxmjd=None, filters=None,
+                               containing_ra=None, containing_dec=None,
+                               minexptime=None, proc_type='raw',
+                               proposals=None ):
+        """Search the NOIRLab data archive for exposures.
+
+        See Instrument.find_origin_exposures for documentation; in addition:
+
+        Parameters
+        ----------
+        filters: str or list of str
+           The short (i.e. single character) filter names ('g', 'r',
+           'i', 'z', or 'Y') to search for.  If not given, will
+           return images from all filters.
+        proposals: str or list of str
+           The NOIRLab proposal ids to limit the search to.  If not
+           given, will not filter based on proposal id.
+        proc_type: str
+           'raw' or 'instcal' : the processing type to get 
+           from the NOIRLab data archive.
```

Comment:
Add missing parameters. Let me just say that using copilot makes this sort of tedium really fun.